ext/magic/extconf.rb

```diff
@@ -1,6 +1,42 @@
 # frozen_string_literal: true
 
 require 'mkmf'
+require 'digest'
+require 'open-uri'
+
+LIBMAGIC_TAG = '5.39'.freeze
+
+workdir = Dir.pwd
+libdir = File.join(workdir, 'file-' + LIBMAGIC_TAG)
+gemdir = File.expand_path(File.join(__dir__, '../..'))
+gem_ext_dir = File.join(gemdir, 'lib', 'ext')
+gem_include_dir = File.join(gem_ext_dir, 'include')
+gem_lib_dir = File.join(gem_ext_dir, 'lib')
+build_lib_dir = File.join(libdir, 'src', '.libs')
+
+expected_sha256 = 'f05d286a76d9556243d0cb05814929c2ecf3a5ba07963f8f70bfaaa70517fad1'
+filename = "#{workdir}/file.tar.gz"
+
+unless File.exist?(filename)
+  File.open(filename, 'wb') do |target_file|
+    URI.open("https://fossies.org/linux/misc/file-#{LIBMAGIC_TAG}.tar.gz", 'rb') do |read_file|
```

Comment:
@stanhu, @kwilczynski  - This line is failing with a 412 error because the `User-Agent` header seems like a mandatory header for FOSSIES.

I've found this by trying the following script on IRB;

```ruby
require 'httparty'

url = "https://fossies.org/linux/misc/file-5.39.tar.gz"
real_agent = "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.82 Safari/537.36"
madeup_agent_1 = "Ruby"
madeup_agent_2 = "Ruby 2.7.0"

HTTParty.get(url).code # => 412
HTTParty.get(url, headers: { "User-Agent" => madeup_agent_1 }).code # => 412

HTTParty.get(url, headers: { "User-Agent" => madeup_agent_2 }).code # => 200
HTTParty.get(url, headers: { "User-Agent" => real_agent }).code # => 200
```

I think we should either include this tarball into the gem source or set the HTTP header as `RUBY_DESCRIPTION`